pkg/networkutils/network.go

```diff
@@ -244,54 +241,129 @@ func (n *linuxNetwork) SetupHostNetwork(vpcCIDR *net.IPNet, primaryMAC string, p
 		return errors.Wrap(err, "host network setup: failed to create iptables")
 	}
 
-	for _, rule := range []iptablesRule{
-		{
-			name:        "connmark for primary ENI",
-			shouldExist: n.nodePortSupportEnabled,
-			table:       "mangle",
-			chain:       "PREROUTING",
-			rule: []string{
-				"-m", "comment", "--comment", "AWS, primary ENI",
-				"-i", "eth0",
-				"-m", "addrtype", "--dst-type", "LOCAL", "--limit-iface-in",
-				"-j", "CONNMARK", "--set-mark", fmt.Sprintf("%#x/%#x", n.mainENIMark, n.mainENIMark),
-			},
-		},
-		{
-			name:        "connmark restore for primary ENI",
-			shouldExist: n.nodePortSupportEnabled,
-			table:       "mangle",
-			chain:       "PREROUTING",
-			rule: []string{
-				"-m", "comment", "--comment", "AWS, primary ENI",
-				"-i", "eni+", "-j", "CONNMARK", "--restore-mark", "--mask", fmt.Sprintf("%#x", n.mainENIMark),
-			},
-		},
-		{
-			name:        fmt.Sprintf("rule for primary address %s", primaryAddr),
+	// build IPTABLES chain for SNAT of non-VPC outbound traffic
+	for i, _ := range vpcCIDRs {
```

Comment:
If this for-loop was from 0 to `len(vpcCIDRs)`, there would be no need for the duplicate code on lines 256-262:
```
	for i := 0; i <= len(vpcCIDRs); i++ {
```